File: shared/src/main/java/com/batu/shared/dto/response/TransactionDto.java
package com.batu.shared.dto.response;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class TransactionDto {

    private UUID transactionId;
    private UUID userId;
    private BigDecimal amount;
    private String isoCurrencyCode;
    private String transactionName;
    private String transactionType;
    private LocalDate date;
    private Boolean pending;
    private String paymentChannel;
    private TransactionDetailedCategoryDto detailedCategory;
    private Instant createdAt;
    private Instant updatedAt;
}

File: transaction-service/src/main/kotlin/com/batu/transaction_service/dto/TransactionDto.kt
package com.batu.transaction_service.dto

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.UUID
import com.batu.transaction_service.dto.TransactionDetailedCategoryDto

data class TransactionDto(
    val transactionId: UUID?,
    val userId: UUID,
    val externalId: String,
    val amount: BigDecimal,
    val isoCurrencyCode: String,
    val transactionName: String,
    val transactionType: String,
    val date: LocalDate,
    val isPending: Boolean?,
    val paymentChannel: String,
    val detailedCategory: TransactionDetailedCategoryDto,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?
)


